File: src/main/kotlin/java/_02_data_calculations_actions/referentialTransparency/Main.java
package java._02_data_calculations_actions.referentialTransparency;

public class Main {

    public static void main(String[] args) {

        System.out.println(square(hypotenuse(square(3), square(4))));
    }

    public static double hypotenuse(double a, double b) {
        var h = Math.sqrt(a * a + b * b);
        return h;
    }

    public static double square(double x) {
        return x * x;
    }
}

File: src/main/kotlin/_02_data_calculations_actions/referentialTransparency/Main.kt
package _02_data_calculations_actions.referentialTransparency

fun hypotenuse(a: Double, b: Double) = Math.sqrt(square(a) + square(b))
fun square(n: Double) = n * n

fun main() {
    println(square(hypotenuse(square(3.0), square(4.0))))
}